### ISSUE TITLE
Committed Multiple files, Received 400

### ISSUE THREAD

**Issue by imrantechwiz** [di-sukharev/opencommit#89]
` Error: Request failed with status code 400`

<img width="866" alt="Screenshot 2023-03-29 at 8 04 21 PM" src="https://user-images.githubusercontent.com/56226186/228572978-26f631b3-a809-48d8-a906-791340c8f547.png">

**Comment by rodrigooler** [comment]
> ` Error: Request failed with status code 400`
> 
> <img alt="Screenshot 2023-03-29 at 8 04 21 PM" width="866" src="https://user-images.githubusercontent.com/56226186/228572978-26f631b3-a809-48d8-a906-791340c8f547.png">

Same problem here, I tested it here in some cases and got this error 400

![Screenshot 2023-03-29 at 1 31 41 PM](https://user-images.githubusercontent.com/5496931/228606618-94feb54d-a9fe-48c2-9c87-bd0abb1f9de2.png)

**Comment by aliharis** [comment]
+1 running into the same issue

**Comment by eudard** [comment]
+1 same error when oc multiple files

**Comment by ProfeJulianLasso** [comment]
+1 running into the same issue

**Comment by rodrigooler** [comment]
Wow, you're doing a great job! @di-sukharev

**Comment by wywywywy** [comment]
> should be fixed with a coming PR, thanks <3

Which PR is it? I still have the same problem with the latest version.

Speaking of latest version, the current 1.1.**36** on npm seems to be published incorrectly. When you install it with `npm i opencommit@1.1.36` and then do `oc --version`, it shows as 1.1.**35**. 

There's probably a mistake made during packaging. Maybe it's time to consider using GitHub Actions.

**Comment by kenryu42** [comment]
Running into the same issues and as @wywywywy said, `oc --version` still shows as `1.1.35`.

**Comment by di-sukharev** [comment]
@wywywywy this is a known issue, idk how to config it.. but anyway the 35 version is same as 36. bug is not fixed, but we diagnosed the issue, gonna merge the PR soon.

**Comment by di-sukharev** [comment]
should be fixed now, pls update `npm i -g opencommit`

**Comment by MartinPaulEve** [comment]
Hmm. With apologies, I updated to 1.1.48 (opencommit --version) and am still getting:

 └  ✖ Error: Request failed with status code 400

**Comment by Kevdome3000** [comment]
same here after running npm i -g opencommit

**Comment by di-sukharev** [comment]
yep, this error persists, could anybody investigate and make a PR?

**Comment by mathsgod** [comment]
One of the possible reasons for this problem is that the token has exceeded the limit of 4096. In the code, result of `git diff` is split and submitted to the openai API by line breaks if over `MAX_REQ_TOKENS`. However, there is a possibility that even after splitting, one sentence still exceeds the limit, which will result in this error.

**Comment by mathsgod** [comment]
You can try the following step to re-create the error:

1. create an empty folder, run `git init`
2. create a text file and write very long string without line breaks (token must be over 4096) , You can go [here](https://platform.openai.com/tokenizer) to check token size.
3. run `oc`

**Comment by jlopezlira** [comment]
Same results with `1.1.50`

**Comment by codeagencybe** [comment]
I also still have the same problem with version 1.1.50/2.0.0, but I can confirm it only happens with "many changes". 
It's not about file size. As soon as I touch many changes to multiple files, at some point it will result in err 400, most likely due to MAX_REQ_TOKENS with limit 4096 from openai API limits. 
If I do small simple commits, it never happens with errors. 

For now, I work around this by doing commits earlier after changes instead of waiting until 1 "larger" commit. 
It's not ideal as multi commits = multiple builds that get triggered, causing more computation expenses etc... 
I would prefer to do only 1 commit where I really need it. 

is there a way to work around that 4096 limit? Can openAI accept bigger? I'm ok with paying more for upgrading something at openai if that solves the problem. But I don't know if something like that is even available. 

I'm happy to help where possible, testing, etc... I just don't where to start or how. 

Thanks.

**Comment by HarryAdney** [comment]
Any update on this? It's pretty useless for me at the moment, tbh.

**Comment by di-sukharev** [comment]
we tried 3 PRs and nothing healed it, if anybody wants to be the HERO who solves this — please do the shot and save us here 🙏 🦺

**Comment by jlopezlira** [comment]
@di-sukharev I'm going to tackle this topic, I'll give it a try.

**Comment by HarryAdney** [comment]
> 

You sir, are a gentleman and a scholar.

**Comment by codeagencybe** [comment]
@di-sukharev 

Does this update from openAI may be helpful to resolve this issue? 
They now seem to support up to 16k tokens at a slightly higher price. 

<img width="679" alt="image" src="https://github.com/di-sukharev/opencommit/assets/7568546/6962da3a-8587-479a-931a-ef76f56e07eb">

**Comment by codeagencybe** [comment]
@di-sukharev why is issue closed as "completed"? 
Because this issue is still existing.

Is this project still ongoing?

**Comment by HarryAdney** [comment]
> @di-sukharev why is issue closed as "completed"? Because this issue is still existing.
> 
> Is this project still ongoing?

No recent messages since your last one on Jun 14, so the github-actions bot closed it.

**Comment by di-sukharev** [comment]
sorry for closing, will add a bigger model context and validate myself
